lib/spack/spack/test/test_activations.py

```diff
@@ -210,9 +210,9 @@ def test_python_ignore_namespace_init_conflict(
     python_spec = spack.spec.Spec('python@2.7.12')
     python_spec._concrete = True
 
-    ext1_pkg = create_python_ext_pkg('py-extension1', ext1_prefix, python_spec,
+    ext1_pkg = create_python_ext_pkg('py-extension1@1.0.0', ext1_prefix, python_spec,
```

Comment:
Why was it necessary to add versions in these tests?

Reply:
Possibly due to the package hashing changes, a bunch of tests had started failing with:

```
lib/spack/spack/test/test_activations.py:224: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
var/spack/repos/builtin/packages/python/package.py:1187: in activate
    ???
lib/spack/spack/package.py:2434: in activate
    view.merge(extension.spec, ignore=kwargs.get('ignore', None))
lib/spack/spack/filesystem_view.py:420: in merge
    pkg.add_files_to_view(self, merge_map)
lib/spack/spack/build_systems/python.py:221: in add_files_to_view
    python_prefix = self.extendee_spec.prefix
lib/spack/spack/spec.py:1493: in prefix
    self.dag_hash())
lib/spack/spack/spec.py:1550: in dag_hash
    return self._cached_hash(ht.dag_hash, length)
lib/spack/spack/spec.py:1535: in _cached_hash
    hash_string = self._spec_hash(hash)
lib/spack/spack/spec.py:1514: in _spec_hash
    node_dict = self.to_node_dict(hash=hash)
lib/spack/spack/spec.py:1666: in to_node_dict
    package_hash = self.package_hash()
lib/spack/spack/spec.py:1543: in package_hash
    return self._cached_hash(ht.package_hash)
lib/spack/spack/spec.py:1535: in _cached_hash
    hash_string = self._spec_hash(hash)
lib/spack/spack/spec.py:1513: in _spec_hash
    return hash.override(self)
lib/spack/spack/hash_types.py:58: in <lambda>
    override=lambda s: s.package.content_hash())
lib/spack/spack/package.py:1553: in content_hash
    source_id = fs.for_package_version(self, self.version).source_id()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <spack.pkg.builtin.python.Python object at 0x7f05cbe00700>

    @property
    def version(self):
        if not self.spec.versions.concrete:
>           raise ValueError("Version requested for a package that"
                             " does not have a concrete version.")
E           ValueError: Version requested for a package that does not have a concrete version.

lib/spack/spack/package.py:897: ValueError
```

Reply:
@becker33 suggested it could be fixed by adding versions where we were missing them, and that turned out to be true.

Reply:
Do we need to understand why this became required?  Or is it sufficient to simply add the versions so the tests pass?

Reply:
I fixed this a different way in 1c054eda7c00c5d5d80cc586273119778367b0d1.  I think we should *not* require tests to mock up versions, especially since nobody understands why we had to.

The reason for this was that `content_hash()` requires a concrete spec — i.e. one with a version. The version hack makes it work here, but it’s better if we just make `content_hash()` best-effort.

Note that I did *not* make every DAG hash on an abstract spec include the content hash — they still don’t.  But IF the content hash is called on a package that doesn’t actually have a concrete spec, it’ll give the best answer it can, and we don’t have to modify these tests to be more complicated than they need to be.

Reply:
@becker33 @alalazo FYI